### ISSUE TITLE
Reinstate z9 / z10 ALS on z Systems

### ISSUE THREAD

**Issue by fjeremic** [eclipse/omr#667]
Reinstate z9 / z10 ALS

Reinstate z9 / z10 architecture level set (ALS) in the OMR z Systems codegen. This is the minimal architecture set which is to be supported by compiler built on top of OMR moving forward. Hence we need to modify all locations that generate instructions introduced in subsequent iterations of  z/Architecture.

**Comment by 0xdaryl** [comment]
@fjeremic : are there any remaining tasks with this issue?  If not, please close it.

**Comment by fjeremic** [comment]
@0xdaryl yes there seems to be one outstanding issue I need to fix. I'm waiting on infrastructure issues to be resolved on the z9 machine so I can hop on and reproduce the problem.
